BPA/Ch.13/Dynocc.stan

```diff
@@ -0,0 +1,98 @@
+// Dynamic (multi-season) site-occupancy model
+// This model is implemented as a hidden Markov model.
+
+// States:
+// 1: occupied
+// 2: not occupied
+// Observations:
+// number of detection + 1
+data {
+  int<lower=1> nsite;                         // Number of sites
+  int<lower=1> nrep;                          // Number of replicate surveys
+  int<lower=1> nyear;                         // Number of years
+  int<lower=0,upper=1> y[nsite, nrep, nyear]; // Detection history
+}
+
+transformed data {
+  int<lower=1,upper=nrep+1> sum_y[nsite, nyear];  // sum + 1
+
+  for (i in 1:nsite)
+    for (k in 1:nyear)
+      sum_y[i, k] <- sum(y[i, 1:nrep, k]) + 1;
+}
+
+parameters {
+  real<lower=0,upper=1> psi1;
+  vector<lower=0,upper=1>[nyear-1] phi;
+  vector<lower=0,upper=1>[nyear-1] gamma;
+  vector<lower=0,upper=1>[nyear] p;
+}
+
+transformed parameters {
+  simplex[2] ps[2, nyear-1];     // Transition probability
+  simplex[nrep+1] po[2, nyear];  // Emission probability
+
+  for (t in 1:(nyear - 1)) {
+    ps[1, t, 1] <- phi[t];
+    ps[1, t, 2] <- 1.0 - phi[t];
+    ps[2, t, 1] <- gamma[t];
+    ps[2, t, 2] <- 1.0 - gamma[t];
+  }
```

Comment:
I tried to transform `po` into `log_po`, but this attempt resulted in sampling error with initial values maybe because `log_po` contained many negative infinities.